### PR TITLE
Make sample show more examples of std::function and std::bind

### PR DESCRIPTION
**Pull request by frankdownunder**

### CODE REVIEW

File: samples/Basic_Delegates/app/application.cpp [hunk 1]
```diff
@@ -1,45 +1,125 @@
...
+Task task4;
+Task task5;
 
 void init()
 {
-	myLed1.setTimer(1000);
-	myLed1.blink(true);
-	myLed2.setTimer(500);
-	myLed2.blink(true);
+	WifiStation.enable(false);
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Remove the Wifi* call on this line and the one below. They are not directly related to the example and to what you are trying to demonstrate.

**Comment by slaff** [comment]
@frankdownunder Before I merge this PR I would like to know if you tested it on a real device?

File: samples/Basic_Delegates/app/application.cpp [hunk 2]
```diff
@@ -1,45 +1,123 @@
...
-	LedBlinker(int reqPin) : ledPin(reqPin) {
-		pinMode(ledPin, OUTPUT);
-	};
+
+class Task
+{
+
+public:
+	Task() {};
 	bool setTimer(int reqInterval) {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

File: samples/Basic_Delegates/app/application.cpp [hunk 3]
```diff
@@ -1,45 +1,123 @@
...
-	void blink(bool reqRun) {
-		if (reqRun) {
-			ledTimer.initializeMs(ledInterval, TimerDelegate(&LedBlinker::ledBlink,this)).start();
-		}
-		else {
-			ledTimer.stop();
-		}
+
+	// This example show the way delegates have been used in Sming in the past.
+	void callOldDelegate() {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

File: samples/Basic_Delegates/app/application.cpp [hunk 4]
```diff
@@ -1,45 +1,123 @@
...
-	void ledBlink () { ledState = !ledState ; digitalWrite(ledPin, ledState);}
 
-private :
-	int ledPin = 2;
-	Timer ledTimer;
-	int ledInterval = 1000;
-	bool ledState = true;
-};
+	// This example shows how to use a plain old ordinary function as a callback
+	void callPlainOldOrdinaryFunction() {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

File: samples/Basic_Delegates/app/application.cpp [hunk 5]
```diff
@@ -1,45 +1,123 @@
...
+		taskTimer.initializeMs(taskInterval, TimerDelegateStdFunction(plainOldOrdinaryFunction)).start();
+		// or just
+		// taskTimer.initializeMs(taskInterval, plainOldOrdinaryFunction).start();
+	}
+
+
+	// This example shows how to use std::bind to make us of a function that has more parameters than our signature has
+	void showHowToUseBind() {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

File: samples/Basic_Delegates/app/application.cpp [hunk 6]
```diff
@@ -1,45 +1,123 @@
...
+		auto b = std::bind(functionWithMoreComlicatedSignature, 2, "parameters");
+		taskTimer.initializeMs(taskInterval, b).start();
+	}
+
+	// Sming now allows the use of std::function
+	// This example shows how to use a lamda expression as a callback
+	void callLamda() {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

File: samples/Basic_Delegates/app/application.cpp [hunk 7]
```diff
@@ -1,45 +1,123 @@
...
+	}
 
-#define LEDPIN_1  2 // GPIO2
-#define LEDPIN_2  4 // GPIO4
 
-LedBlinker myLed1 = LedBlinker(LEDPIN_1);
-LedBlinker myLed2 = LedBlinker(LEDPIN_2);
+
+	// This example shows how to use a member function as a callback
+	void callMemberFunction() {
```
Comment (slaff) [inline samples/Basic_Delegates/app/application.cpp]: Coding-Style: Put the opening curly brace on the next line.

**Comment by frankdownunder** [comment]
@slaff   I built this sample using Ubuntu from the Windows Store (the replacement for WSL on Windows 10). To build it, I had to add these lines into makefile-user.mk (which I did not commit)
## size of the flash chip
SPI_SIZE        ?= 4M
COM_PORT = /dev/ttyS7

I tested it on a Wemos D1 mini by running it for a couple of minutes.

**Comment by slaff** [comment]
> I am new to git, I find it very confusing ☹

It is not that difficult ;-) . Go to your branch: feature/DelegatorsAndStdFunction_1337x2
```
git checkout  feature/DelegatorsAndStdFunction_1337x2
```
Then make sure that you have commited your changes and finally do
```
git push
```
So that those changes show up also in the PR. 
Thanks in advance.

**Comment by frankdownunder** [comment]
Try now


Sent from Mail for Windows 10

**Pull request merged**